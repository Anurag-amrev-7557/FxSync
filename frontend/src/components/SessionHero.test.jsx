import React from 'react';
import { render, screen } from '@testing-library/react';
import SessionHero from './SessionHero';

describe('SessionHero', () => {
  it('renders and matches snapshot when visible', () => {
    const { asFragment } = render(<SessionHero isVisible={true} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders and matches snapshot when not visible', () => {
    const { asFragment } = render(<SessionHero isVisible={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders all feature highlights', () => {
    render(<SessionHero isVisible={true} />);
    expect(screen.getByLabelText(/real-time sync/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/group chat/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/playlist sharing/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/no account needed/i)).toBeInTheDocument();
  });

  it('has accessible tooltips for all features', () => {
    render(<SessionHero isVisible={true} />);
    expect(screen.getByRole('tooltip', { name: /everyone hears the same thing/i })).toBeInTheDocument();
    expect(screen.getByRole('tooltip', { name: /chat live with everyone/i })).toBeInTheDocument();
    expect(screen.getByRole('tooltip', { name: /collaborate on the perfect queue/i })).toBeInTheDocument();
    expect(screen.getByRole('tooltip', { name: /jump in instantly/i })).toBeInTheDocument();
  });
}); 