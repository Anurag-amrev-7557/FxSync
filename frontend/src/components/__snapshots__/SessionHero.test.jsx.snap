// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SessionHero renders and matches snapshot when not visible 1`] = `
<DocumentFragment>
  <div
    class="text-center lg:text-left space-y-8 sm:space-y-10 order-1 lg:order-1 mb-8 sm:mb-0 relative"
  >
    <div
      class="pointer-events-none absolute -top-10 -left-10 w-40 h-40 opacity-40 blur-2xl z-0 animate-float-slow"
    >
      <svg
        fill="none"
        viewBox="0 0 200 200"
      >
        <ellipse
          cx="100"
          cy="100"
          fill="#60a5fa"
          fill-opacity="0.25"
          rx="90"
          ry="60"
        />
      </svg>
    </div>
    <div
      class="pointer-events-none absolute -bottom-12 -right-12 w-32 h-32 opacity-30 blur-2xl z-0 animate-float-slower"
    >
      <svg
        fill="none"
        viewBox="0 0 200 200"
      >
        <ellipse
          cx="100"
          cy="100"
          fill="#f472b6"
          fill-opacity="0.18"
          rx="80"
          ry="50"
        />
      </svg>
    </div>
    <div
      class="space-y-4 sm:space-y-6 relative z-10"
    >
      <h1
        class="text-3xl sm:text-4xl md:text-5xl lg:text-7xl font-extrabold leading-tight text-white transition-all duration-1000 delay-300 drop-shadow-lg opacity-0 translate-y-8"
      >
        <span
          class="inline-block transition-all duration-700 delay-500 hover:scale-110 hover:text-blue-200 cursor-pointer group relative overflow-visible"
        >
          <span
            class="relative z-10 bg-gradient-to-r from-white via-blue-100 to-neutral-200 bg-clip-text text-transparent group-hover:from-blue-200 group-hover:to-white transition-all duration-500"
          >
            Sync Your
          </span>
          <div
            class="absolute inset-0 bg-gradient-to-r from-transparent via-blue-400/20 to-transparent transform -skew-x-12 transition-transform duration-700 group-hover:translate-x-full"
          />
          <svg
            class="absolute -top-4 -right-8 w-8 h-8 text-blue-300 opacity-0 group-hover:opacity-100 transition-all duration-500 group-hover:animate-bounce"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 18V5l12-2v13"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <circle
              cx="18"
              cy="16"
              r="3"
            />
          </svg>
          <svg
            class="absolute -top-2 left-1/2 -translate-x-1/2 w-4 h-4 text-yellow-200 opacity-0 group-hover:opacity-100 transition-all duration-700 group-hover:animate-twinkle"
            fill="currentColor"
            viewBox="0 0 20 20"
          >
            <polygon
              points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
            />
          </svg>
        </span>
        <br />
        <span
          class="text-neutral-300 inline-block transition-all duration-700 delay-700 hover:scale-110 hover:text-pink-200 cursor-pointer group relative overflow-visible"
        >
          <span
            class="relative z-10 bg-gradient-to-r from-neutral-300 via-pink-200 to-white bg-clip-text text-transparent group-hover:from-white group-hover:to-pink-200 transition-all duration-500"
          >
            Music
          </span>
          <div
            class="absolute inset-0 bg-gradient-to-r from-transparent via-pink-400/30 to-transparent transform -skew-x-12 transition-transform duration-700 group-hover:translate-x-full"
          />
          <div
            class="absolute -inset-1 bg-gradient-to-r from-pink-400/20 via-neutral-500/30 to-blue-400/20 rounded-lg blur opacity-0 group-hover:opacity-100 transition-opacity duration-500"
          />
          <div
            class="absolute -bottom-2 left-0 flex items-end gap-0.5 opacity-0 group-hover:opacity-100 transition-all duration-500"
          >
            <div
              class="w-0.5 bg-gradient-to-t from-pink-300 to-white rounded-full animate-pulse"
              style="height: 6px; animation-delay: 0ms; animation-duration: 1.2s;"
            />
            <div
              class="w-0.5 bg-gradient-to-t from-pink-300 to-white rounded-full animate-pulse"
              style="height: 12px; animation-delay: 120ms; animation-duration: 1.2s;"
            />
            <div
              class="w-0.5 bg-gradient-to-t from-pink-300 to-white rounded-full animate-pulse"
              style="height: 18px; animation-delay: 240ms; animation-duration: 1.2s;"
            />
            <div
              class="w-0.5 bg-gradient-to-t from-pink-300 to-white rounded-full animate-pulse"
              style="height: 12px; animation-delay: 360ms; animation-duration: 1.2s;"
            />
            <div
              class="w-0.5 bg-gradient-to-t from-pink-300 to-white rounded-full animate-pulse"
              style="height: 6px; animation-delay: 480ms; animation-duration: 1.2s;"
            />
          </div>
          <svg
            class="absolute -top-4 -left-4 w-5 h-5 text-pink-300 opacity-0 group-hover:opacity-100 transition-all duration-700 group-hover:animate-spin"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 18V5l12-2v13"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <circle
              cx="18"
              cy="16"
              r="3"
            />
          </svg>
          <svg
            class="absolute -bottom-3 -right-3 w-4 h-4 text-blue-400 opacity-0 group-hover:opacity-100 transition-all duration-700 delay-200 group-hover:animate-ping"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 18V5l12-2v13"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <circle
              cx="18"
              cy="16"
              r="3"
            />
          </svg>
        </span>
      </h1>
      <p
        class="text-base sm:text-lg lg:text-xl text-neutral-300 max-w-lg mx-auto lg:mx-0 leading-relaxed transition-all duration-1000 delay-900 drop-shadow opacity-0 translate-y-8"
      >
        Create 
        <span
          class="font-semibold text-white"
        >
          synchronized listening rooms
        </span>
         where everyone experiences music together.
        <span
          class="inline-block ml-2 transition-all duration-300 hover:rotate-12 animate-wiggle"
        >
          ðŸŽ§
        </span>
      </p>
    </div>
    <div
      class="flex flex-wrap gap-4 sm:gap-6 max-w-md mx-auto lg:mx-0 justify-center lg:justify-start transition-all duration-1000 delay-1100 opacity-0 translate-y-8"
    >
      <div
        aria-describedby=":r4:"
        aria-label="Real-time sync"
        class="relative flex items-center gap-2 text-xs sm:text-sm text-neutral-400 group cursor-pointer transition-all duration-300 hover:text-white hover:scale-105"
        role="button"
        style="transition-delay: 0ms;"
        tabindex="0"
      >
        <div
          class="flex items-center justify-center w-8 h-8 rounded-full bg-gradient-to-br from-blue-400/30 to-blue-200/10 transition-all duration-300 shadow-md"
        >
          <svg
            class="w-5 h-5 text-white group-hover:text-blue-200 transition-colors duration-300 drop-shadow"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 12a9 9 0 1 1-3.5-7"
            />
            <polyline
              points="21 3 21 8 16 8"
            />
          </svg>
        </div>
        <span
          class="transition-all duration-300 group-hover:translate-x-1 font-semibold"
        >
          Real-time sync
        </span>
        <div
          aria-live="polite"
          class="absolute left-1/2 -translate-x-1/2 top-full mt-2 z-20 opacity-0 group-hover:opacity-100 group-focus:opacity-100 pointer-events-none transition-opacity duration-300"
          id=":r4:"
          role="tooltip"
        >
          <span
            class="px-2 py-1 rounded bg-neutral-900/90 text-xs text-neutral-200 shadow-lg border border-neutral-700 whitespace-nowrap"
          >
            Everyone hears the same thing at the same time
          </span>
        </div>
        <svg
          class="absolute -top-2 -right-2 w-3 h-3 text-white opacity-0 group-hover:opacity-80 transition-all duration-500 group-hover:animate-twinkle"
          fill="currentColor"
          viewBox="0 0 20 20"
        >
          <polygon
            points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
          />
        </svg>
      </div>
      <div
        aria-describedby=":r5:"
        aria-label="Group chat"
        class="relative flex items-center gap-2 text-xs sm:text-sm text-neutral-400 group cursor-pointer transition-all duration-300 hover:text-white hover:scale-105"
        role="button"
        style="transition-delay: 100ms;"
        tabindex="0"
      >
        <div
          class="flex items-center justify-center w-8 h-8 rounded-full bg-gradient-to-br from-green-400/30 to-green-200/10 transition-all duration-300 shadow-md"
        >
          <svg
            class="w-5 h-5 text-white group-hover:text-green-200 transition-colors duration-300 drop-shadow"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 15a2 2 0 0 1-2 2H7l-4 4V5a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2z"
            />
          </svg>
        </div>
        <span
          class="transition-all duration-300 group-hover:translate-x-1 font-semibold"
        >
          Group chat
        </span>
        <div
          aria-live="polite"
          class="absolute left-1/2 -translate-x-1/2 top-full mt-2 z-20 opacity-0 group-hover:opacity-100 group-focus:opacity-100 pointer-events-none transition-opacity duration-300"
          id=":r5:"
          role="tooltip"
        >
          <span
            class="px-2 py-1 rounded bg-neutral-900/90 text-xs text-neutral-200 shadow-lg border border-neutral-700 whitespace-nowrap"
          >
            Chat live with everyone in the room
          </span>
        </div>
        <svg
          class="absolute -top-2 -right-2 w-3 h-3 text-white opacity-0 group-hover:opacity-80 transition-all duration-500 group-hover:animate-twinkle"
          fill="currentColor"
          viewBox="0 0 20 20"
        >
          <polygon
            points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
          />
        </svg>
      </div>
      <div
        aria-describedby=":r6:"
        aria-label="Playlist sharing"
        class="relative flex items-center gap-2 text-xs sm:text-sm text-neutral-400 group cursor-pointer transition-all duration-300 hover:text-white hover:scale-105"
        role="button"
        style="transition-delay: 200ms;"
        tabindex="0"
      >
        <div
          class="flex items-center justify-center w-8 h-8 rounded-full bg-gradient-to-br from-pink-400/30 to-pink-200/10 transition-all duration-300 shadow-md"
        >
          <svg
            class="w-5 h-5 text-white group-hover:text-pink-200 transition-colors duration-300 drop-shadow"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <rect
              height="18"
              rx="2"
              width="18"
              x="3"
              y="3"
            />
            <path
              d="M9 9h6v6H9z"
            />
          </svg>
        </div>
        <span
          class="transition-all duration-300 group-hover:translate-x-1 font-semibold"
        >
          Playlist sharing
        </span>
        <div
          aria-live="polite"
          class="absolute left-1/2 -translate-x-1/2 top-full mt-2 z-20 opacity-0 group-hover:opacity-100 group-focus:opacity-100 pointer-events-none transition-opacity duration-300"
          id=":r6:"
          role="tooltip"
        >
          <span
            class="px-2 py-1 rounded bg-neutral-900/90 text-xs text-neutral-200 shadow-lg border border-neutral-700 whitespace-nowrap"
          >
            Collaborate on the perfect queue
          </span>
        </div>
        <svg
          class="absolute -top-2 -right-2 w-3 h-3 text-white opacity-0 group-hover:opacity-80 transition-all duration-500 group-hover:animate-twinkle"
          fill="currentColor"
          viewBox="0 0 20 20"
        >
          <polygon
            points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
          />
        </svg>
      </div>
      <div
        aria-describedby=":r7:"
        aria-label="No account needed"
        class="relative flex items-center gap-2 text-xs sm:text-sm text-neutral-400 group cursor-pointer transition-all duration-300 hover:text-white hover:scale-105"
        role="button"
        style="transition-delay: 300ms;"
        tabindex="0"
      >
        <div
          class="flex items-center justify-center w-8 h-8 rounded-full bg-gradient-to-br from-yellow-400/30 to-yellow-200/10 transition-all duration-300 shadow-md"
        >
          <svg
            class="w-5 h-5 text-white group-hover:text-yellow-200 transition-colors duration-300 drop-shadow"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="M15 9l-6 6M9 9l6 6"
            />
          </svg>
        </div>
        <span
          class="transition-all duration-300 group-hover:translate-x-1 font-semibold"
        >
          No account needed
        </span>
        <div
          aria-live="polite"
          class="absolute left-1/2 -translate-x-1/2 top-full mt-2 z-20 opacity-0 group-hover:opacity-100 group-focus:opacity-100 pointer-events-none transition-opacity duration-300"
          id=":r7:"
          role="tooltip"
        >
          <span
            class="px-2 py-1 rounded bg-neutral-900/90 text-xs text-neutral-200 shadow-lg border border-neutral-700 whitespace-nowrap"
          >
            Jump in instantly, no sign up
          </span>
        </div>
        <svg
          class="absolute -top-2 -right-2 w-3 h-3 text-white opacity-0 group-hover:opacity-80 transition-all duration-500 group-hover:animate-twinkle"
          fill="currentColor"
          viewBox="0 0 20 20"
        >
          <polygon
            points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SessionHero renders and matches snapshot when visible 1`] = `
<DocumentFragment>
  <div
    class="text-center lg:text-left space-y-8 sm:space-y-10 order-1 lg:order-1 mb-8 sm:mb-0 relative"
  >
    <div
      class="pointer-events-none absolute -top-10 -left-10 w-40 h-40 opacity-40 blur-2xl z-0 animate-float-slow"
    >
      <svg
        fill="none"
        viewBox="0 0 200 200"
      >
        <ellipse
          cx="100"
          cy="100"
          fill="#60a5fa"
          fill-opacity="0.25"
          rx="90"
          ry="60"
        />
      </svg>
    </div>
    <div
      class="pointer-events-none absolute -bottom-12 -right-12 w-32 h-32 opacity-30 blur-2xl z-0 animate-float-slower"
    >
      <svg
        fill="none"
        viewBox="0 0 200 200"
      >
        <ellipse
          cx="100"
          cy="100"
          fill="#f472b6"
          fill-opacity="0.18"
          rx="80"
          ry="50"
        />
      </svg>
    </div>
    <div
      class="space-y-4 sm:space-y-6 relative z-10"
    >
      <h1
        class="text-3xl sm:text-4xl md:text-5xl lg:text-7xl font-extrabold leading-tight text-white transition-all duration-1000 delay-300 drop-shadow-lg opacity-100 translate-y-0"
      >
        <span
          class="inline-block transition-all duration-700 delay-500 hover:scale-110 hover:text-blue-200 cursor-pointer group relative overflow-visible"
        >
          <span
            class="relative z-10 bg-gradient-to-r from-white via-blue-100 to-neutral-200 bg-clip-text text-transparent group-hover:from-blue-200 group-hover:to-white transition-all duration-500"
          >
            Sync Your
          </span>
          <div
            class="absolute inset-0 bg-gradient-to-r from-transparent via-blue-400/20 to-transparent transform -skew-x-12 transition-transform duration-700 group-hover:translate-x-full"
          />
          <svg
            class="absolute -top-4 -right-8 w-8 h-8 text-blue-300 opacity-0 group-hover:opacity-100 transition-all duration-500 group-hover:animate-bounce"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 18V5l12-2v13"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <circle
              cx="18"
              cy="16"
              r="3"
            />
          </svg>
          <svg
            class="absolute -top-2 left-1/2 -translate-x-1/2 w-4 h-4 text-yellow-200 opacity-0 group-hover:opacity-100 transition-all duration-700 group-hover:animate-twinkle"
            fill="currentColor"
            viewBox="0 0 20 20"
          >
            <polygon
              points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
            />
          </svg>
        </span>
        <br />
        <span
          class="text-neutral-300 inline-block transition-all duration-700 delay-700 hover:scale-110 hover:text-pink-200 cursor-pointer group relative overflow-visible"
        >
          <span
            class="relative z-10 bg-gradient-to-r from-neutral-300 via-pink-200 to-white bg-clip-text text-transparent group-hover:from-white group-hover:to-pink-200 transition-all duration-500"
          >
            Music
          </span>
          <div
            class="absolute inset-0 bg-gradient-to-r from-transparent via-pink-400/30 to-transparent transform -skew-x-12 transition-transform duration-700 group-hover:translate-x-full"
          />
          <div
            class="absolute -inset-1 bg-gradient-to-r from-pink-400/20 via-neutral-500/30 to-blue-400/20 rounded-lg blur opacity-0 group-hover:opacity-100 transition-opacity duration-500"
          />
          <div
            class="absolute -bottom-2 left-0 flex items-end gap-0.5 opacity-0 group-hover:opacity-100 transition-all duration-500"
          >
            <div
              class="w-0.5 bg-gradient-to-t from-pink-300 to-white rounded-full animate-pulse"
              style="height: 6px; animation-delay: 0ms; animation-duration: 1.2s;"
            />
            <div
              class="w-0.5 bg-gradient-to-t from-pink-300 to-white rounded-full animate-pulse"
              style="height: 12px; animation-delay: 120ms; animation-duration: 1.2s;"
            />
            <div
              class="w-0.5 bg-gradient-to-t from-pink-300 to-white rounded-full animate-pulse"
              style="height: 18px; animation-delay: 240ms; animation-duration: 1.2s;"
            />
            <div
              class="w-0.5 bg-gradient-to-t from-pink-300 to-white rounded-full animate-pulse"
              style="height: 12px; animation-delay: 360ms; animation-duration: 1.2s;"
            />
            <div
              class="w-0.5 bg-gradient-to-t from-pink-300 to-white rounded-full animate-pulse"
              style="height: 6px; animation-delay: 480ms; animation-duration: 1.2s;"
            />
          </div>
          <svg
            class="absolute -top-4 -left-4 w-5 h-5 text-pink-300 opacity-0 group-hover:opacity-100 transition-all duration-700 group-hover:animate-spin"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 18V5l12-2v13"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <circle
              cx="18"
              cy="16"
              r="3"
            />
          </svg>
          <svg
            class="absolute -bottom-3 -right-3 w-4 h-4 text-blue-400 opacity-0 group-hover:opacity-100 transition-all duration-700 delay-200 group-hover:animate-ping"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 18V5l12-2v13"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <circle
              cx="18"
              cy="16"
              r="3"
            />
          </svg>
        </span>
      </h1>
      <p
        class="text-base sm:text-lg lg:text-xl text-neutral-300 max-w-lg mx-auto lg:mx-0 leading-relaxed transition-all duration-1000 delay-900 drop-shadow opacity-100 translate-y-0"
      >
        Create 
        <span
          class="font-semibold text-white"
        >
          synchronized listening rooms
        </span>
         where everyone experiences music together.
        <span
          class="inline-block ml-2 transition-all duration-300 hover:rotate-12 animate-wiggle"
        >
          ðŸŽ§
        </span>
      </p>
    </div>
    <div
      class="flex flex-wrap gap-4 sm:gap-6 max-w-md mx-auto lg:mx-0 justify-center lg:justify-start transition-all duration-1000 delay-1100 opacity-100 translate-y-0"
    >
      <div
        aria-describedby=":r0:"
        aria-label="Real-time sync"
        class="relative flex items-center gap-2 text-xs sm:text-sm text-neutral-400 group cursor-pointer transition-all duration-300 hover:text-white hover:scale-105"
        role="button"
        style="transition-delay: 0ms;"
        tabindex="0"
      >
        <div
          class="flex items-center justify-center w-8 h-8 rounded-full bg-gradient-to-br from-blue-400/30 to-blue-200/10 transition-all duration-300 shadow-md"
        >
          <svg
            class="w-5 h-5 text-white group-hover:text-blue-200 transition-colors duration-300 drop-shadow"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 12a9 9 0 1 1-3.5-7"
            />
            <polyline
              points="21 3 21 8 16 8"
            />
          </svg>
        </div>
        <span
          class="transition-all duration-300 group-hover:translate-x-1 font-semibold"
        >
          Real-time sync
        </span>
        <div
          aria-live="polite"
          class="absolute left-1/2 -translate-x-1/2 top-full mt-2 z-20 opacity-0 group-hover:opacity-100 group-focus:opacity-100 pointer-events-none transition-opacity duration-300"
          id=":r0:"
          role="tooltip"
        >
          <span
            class="px-2 py-1 rounded bg-neutral-900/90 text-xs text-neutral-200 shadow-lg border border-neutral-700 whitespace-nowrap"
          >
            Everyone hears the same thing at the same time
          </span>
        </div>
        <svg
          class="absolute -top-2 -right-2 w-3 h-3 text-white opacity-0 group-hover:opacity-80 transition-all duration-500 group-hover:animate-twinkle"
          fill="currentColor"
          viewBox="0 0 20 20"
        >
          <polygon
            points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
          />
        </svg>
      </div>
      <div
        aria-describedby=":r1:"
        aria-label="Group chat"
        class="relative flex items-center gap-2 text-xs sm:text-sm text-neutral-400 group cursor-pointer transition-all duration-300 hover:text-white hover:scale-105"
        role="button"
        style="transition-delay: 100ms;"
        tabindex="0"
      >
        <div
          class="flex items-center justify-center w-8 h-8 rounded-full bg-gradient-to-br from-green-400/30 to-green-200/10 transition-all duration-300 shadow-md"
        >
          <svg
            class="w-5 h-5 text-white group-hover:text-green-200 transition-colors duration-300 drop-shadow"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 15a2 2 0 0 1-2 2H7l-4 4V5a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2z"
            />
          </svg>
        </div>
        <span
          class="transition-all duration-300 group-hover:translate-x-1 font-semibold"
        >
          Group chat
        </span>
        <div
          aria-live="polite"
          class="absolute left-1/2 -translate-x-1/2 top-full mt-2 z-20 opacity-0 group-hover:opacity-100 group-focus:opacity-100 pointer-events-none transition-opacity duration-300"
          id=":r1:"
          role="tooltip"
        >
          <span
            class="px-2 py-1 rounded bg-neutral-900/90 text-xs text-neutral-200 shadow-lg border border-neutral-700 whitespace-nowrap"
          >
            Chat live with everyone in the room
          </span>
        </div>
        <svg
          class="absolute -top-2 -right-2 w-3 h-3 text-white opacity-0 group-hover:opacity-80 transition-all duration-500 group-hover:animate-twinkle"
          fill="currentColor"
          viewBox="0 0 20 20"
        >
          <polygon
            points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
          />
        </svg>
      </div>
      <div
        aria-describedby=":r2:"
        aria-label="Playlist sharing"
        class="relative flex items-center gap-2 text-xs sm:text-sm text-neutral-400 group cursor-pointer transition-all duration-300 hover:text-white hover:scale-105"
        role="button"
        style="transition-delay: 200ms;"
        tabindex="0"
      >
        <div
          class="flex items-center justify-center w-8 h-8 rounded-full bg-gradient-to-br from-pink-400/30 to-pink-200/10 transition-all duration-300 shadow-md"
        >
          <svg
            class="w-5 h-5 text-white group-hover:text-pink-200 transition-colors duration-300 drop-shadow"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <rect
              height="18"
              rx="2"
              width="18"
              x="3"
              y="3"
            />
            <path
              d="M9 9h6v6H9z"
            />
          </svg>
        </div>
        <span
          class="transition-all duration-300 group-hover:translate-x-1 font-semibold"
        >
          Playlist sharing
        </span>
        <div
          aria-live="polite"
          class="absolute left-1/2 -translate-x-1/2 top-full mt-2 z-20 opacity-0 group-hover:opacity-100 group-focus:opacity-100 pointer-events-none transition-opacity duration-300"
          id=":r2:"
          role="tooltip"
        >
          <span
            class="px-2 py-1 rounded bg-neutral-900/90 text-xs text-neutral-200 shadow-lg border border-neutral-700 whitespace-nowrap"
          >
            Collaborate on the perfect queue
          </span>
        </div>
        <svg
          class="absolute -top-2 -right-2 w-3 h-3 text-white opacity-0 group-hover:opacity-80 transition-all duration-500 group-hover:animate-twinkle"
          fill="currentColor"
          viewBox="0 0 20 20"
        >
          <polygon
            points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
          />
        </svg>
      </div>
      <div
        aria-describedby=":r3:"
        aria-label="No account needed"
        class="relative flex items-center gap-2 text-xs sm:text-sm text-neutral-400 group cursor-pointer transition-all duration-300 hover:text-white hover:scale-105"
        role="button"
        style="transition-delay: 300ms;"
        tabindex="0"
      >
        <div
          class="flex items-center justify-center w-8 h-8 rounded-full bg-gradient-to-br from-yellow-400/30 to-yellow-200/10 transition-all duration-300 shadow-md"
        >
          <svg
            class="w-5 h-5 text-white group-hover:text-yellow-200 transition-colors duration-300 drop-shadow"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="M15 9l-6 6M9 9l6 6"
            />
          </svg>
        </div>
        <span
          class="transition-all duration-300 group-hover:translate-x-1 font-semibold"
        >
          No account needed
        </span>
        <div
          aria-live="polite"
          class="absolute left-1/2 -translate-x-1/2 top-full mt-2 z-20 opacity-0 group-hover:opacity-100 group-focus:opacity-100 pointer-events-none transition-opacity duration-300"
          id=":r3:"
          role="tooltip"
        >
          <span
            class="px-2 py-1 rounded bg-neutral-900/90 text-xs text-neutral-200 shadow-lg border border-neutral-700 whitespace-nowrap"
          >
            Jump in instantly, no sign up
          </span>
        </div>
        <svg
          class="absolute -top-2 -right-2 w-3 h-3 text-white opacity-0 group-hover:opacity-80 transition-all duration-500 group-hover:animate-twinkle"
          fill="currentColor"
          viewBox="0 0 20 20"
        >
          <polygon
            points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
